cpp/arcticdb/storage/storage_factory.hpp

```diff
@@ -15,7 +15,10 @@
 
 namespace arcticdb {
     namespace storage {
-
+        std::shared_ptr<Storage> create_storage(
+                const LibraryPath &library_path,
+                OpenMode mode,
+                const NativeVariantStorage& storage_descriptor);
         std::shared_ptr<Storage> create_storage(
```

Comment:
I don't think we should have overloads here. There should be a single `create_storage` method that takes a variant that's either the native object (for S3) or a protobuf (for the other storages). I would also be fine with just biting the bullet now and moving the other storages over to native configuration within the code too if you feel that would help - I'm wondering whether doing this incrementally is adding more complexity than it's worth.

Reply:
Yes it is for doing the shift from proto to native config incrementally. Storage config other than `S3` is still untouched. 
Eventually the proto version of `create_storages` will be removed